### PR TITLE
38 sprint 2 convert home page form to flask wtforms

### PR DESCRIPTION
**Pull request by ipr17**
As you can see from my commits: 
Converted the home page form and ajax form to flask-wtf form
Added the ability to close the photo consent popup and cancel form submission
Added a submit button to the student sign in search bar

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-from app.forms import SessionForm
+from app.forms import LoginForm, SessionForm, StudentSignInForm
 
 
 
 # HOME -   /home/
-@app.route('/', methods=['GET'])
-@app.route('/home', methods=['GET'])
+@app.route('/', methods=['GET', 'POST'])
+@app.route('/home', methods=['GET', 'POST'])
```
Comment (DeltaO3) [inline app/routes.py:10]: Personally am a fan of seperating the route for the POST request into a different route (e.g. /addstudent or something similarly named)- redirect occurs anyway, neater, makes more sense, seperates functionality, decreases complexity of one route etc. This is a matter of personal preference however. Can just pass the form to the different route. Granted, I haven't tested it that way, so perhaps something about the way the JS and forms interact could cause a mess with this.

**Review by DeltaO3 — COMMENTED** [review]
Overall looks great - some minor comments I'd like you to check. Main thing is whether we want to seperate the route - I'm personally a fan of doing the POST seperate to the main GET - especially considering this is the main page and may down the line have more things added to it.

File: app/static/js/consent.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 //Runs when attendance form is submitted - put post request to flask in here
 $("#attendanceForm").submit(function (e) {
+	e.preventDefault();  // Prevent the default form submission
```
[the remark this answers is not in the document]
Comment (ipr17) [reply app/static/js/consent.js:3]: yeah i was just encountering some bugs before so i added that in as well as a check, and forgot to remove it

**Review by DeltaO3 — APPROVED** [review]
LGTM!

**Review by livvy10 — APPROVED** [review]
Works well, nice styling and layout etc. 
I like being able to cancel a student sign in by cancelling the consent modal. But consider if the unit is configured to not ask about consent, so no modal appears - something to think about.

**Pull request merged**